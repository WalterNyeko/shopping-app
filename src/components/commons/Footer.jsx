import React from "react";
import { NavLink } from "react-router-dom";

const Footer = ({ departments }) => {
  return (
    <footer className="footer bg-dark sticky-bottom mb-auto py-3">
      <div className="container text-center h5 mt-4">
        {departments &&
          departments.length &&
          departments.map(({ name, department_id }) => (
            <NavLink
              to={`/department/${department_id}`}
              key={department_id}
              className="text-white mr-5"
            >
              {name}
            </NavLink>
          ))}
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-instagram fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-pinterest fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-twitter fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-facebook fa-3x" />
        </a>
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          &copy; 2019 Shopmate Ltd
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Contact
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Privacy policy
        </a>
      </div>
    </footer>
  );
};
export default Footer;
