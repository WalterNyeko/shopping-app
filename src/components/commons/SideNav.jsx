import React, { Fragment } from 'react';
import '../../styles/SideNav.css';

 const SideNav = (props) => {
     const { content } = props;
    return (
        <Fragment>
            <div className="row">
                <div className="col-md-2">
                    <div className="sidenav">
                        <h6>Categories</h6>
                        <hr></hr>
                        <a href="#">Christmas</a>
                        <a href="#">Valentines</a>
                        <a href="#">Ladies</a>
                        <a href="#">Children</a>
                    </div>
                </div>
                <div className="col-md-10">
                    {content}
                </div>

            </div>
        </Fragment>
    )
}
export default SideNav;
