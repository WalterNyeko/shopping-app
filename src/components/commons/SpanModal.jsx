import React, { Fragment } from 'react';
import { Modal} from 'antd';



const ReusableModal = (props) => {
    const { 
        spanText, 
        modalTitle, 
        modalContent, 
        modalWidth,
        handleOk, 
        handleCancel,
        showModal,
        visible,
        confirmLoading,
        className,
    } = props;
    return (
      <Fragment>
        <span onClick={showModal} className={className}>
            {spanText}
        </span>
        <Modal
            title={modalTitle}
            visible={visible}
            onOk={handleOk}
            confirmLoading={confirmLoading}
            onCancel={handleCancel}
            width={modalWidth}
            footer={null}
        >
            <p>{modalContent}</p>
        </Modal>
      </Fragment>
    );
}
export default ReusableModal;