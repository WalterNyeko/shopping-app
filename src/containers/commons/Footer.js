import React, { Component, Fragment } from "react";
import FooterPage from "../../components/commons/Footer";

class Footer extends Component {
  render() {
    const { departments } = this.props;
    return (
      <Fragment>
        <FooterPage departments={departments} />
      </Fragment>
    );
  }
}
export default Footer;
