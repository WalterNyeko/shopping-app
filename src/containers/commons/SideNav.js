import React, { Component, Fragment } from 'react';
import SideNavComponent from '../../components/commons/SideNav';

class SideNav extends Component {
    render() {
        const { content } = this.props;
        return (
            <Fragment>
                <SideNavComponent content={content}/>
            </Fragment>
        )
    }
}
export default SideNav;
