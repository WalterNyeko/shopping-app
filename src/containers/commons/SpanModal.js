import React, { Component, Fragment } from "react";
import SpanModalComponent from "../../components/commons/SpanModal";

export class SpanModal extends Component {
  state = {
    visible: false,
    confirmLoading: false
  };

  /**
   * triggers the modal to show up
   *
   * @returns {void}
   */
  showModal = () => {
    this.setState({
      visible: true
    });
  };

  /**
   * ensures that a modal state, visible is set to false
   * this therefore allows modal to hide
   *
   * @returns {void}
   */
  handleOk = () => {
    this.setState({
      visible: false
    });
  };

  /**
   * triggers the modal to hide
   * sets the state of the modal, visible to false
   *
   * @returns {void}
   */
  handleCancel = () => {
    this.setState({
      visible: false
    });
  };

  render() {
    const { visible, confirmLoading } = this.state;
    const {
      spanText,
      modalTitle,
      modalContent,
      modalWidth,
      className
    } = this.props;
    return (
      <Fragment>
        <SpanModalComponent
          visible={visible}
          confirmLoading={confirmLoading}
          spanText={spanText}
          modalTitle={modalTitle}
          modalContent={modalContent}
          modalWidth={modalWidth}
          showModal={this.showModal}
          handleCancel={this.handleCancel}
          handleOk={this.handleOk}
          className={className}
        />
      </Fragment>
    );
  }
}
