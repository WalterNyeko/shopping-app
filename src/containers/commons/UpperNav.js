import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import UpperNavComponent from "../../components/commons/UpperNav";
import { getDepartments } from "../../store/actions/Departments";
import { getTotalAmount } from "../../store/actions/Orders";

class UpperNav extends Component {
  state = {
    isLoggedIn: false
  };

  componentWillMount() {
    const jwtToken = localStorage.getItem("jwt-token");
    const { getDepartments, getTotalAmount, getCartUniqueId } = this.props;
    jwtToken && this.setState({ isLoggedIn: true });
    getDepartments();
    const cartId = localStorage.getItem("cartId");
    getTotalAmount(cartId);
  }
  render() {
    const { isLoggedIn } = this.state;
    const {
      allDepartments,
      myCart: {
        totalAmountOfItemsInCart: { total_amount }
      }
    } = this.props;
    return (
      <Fragment>
        <UpperNavComponent
          isLoggedIn={isLoggedIn}
          allDepartments={allDepartments}
          totalAmount={total_amount}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allDepartments: state.departments,
  myCart: state.itemsInCart
});
export default connect(
  mapStateToProps,
  { getDepartments, getTotalAmount }
)(UpperNav);
