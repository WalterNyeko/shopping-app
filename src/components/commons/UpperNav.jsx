import React, { Fragment } from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { NavLink } from "react-router-dom";
import bag from "../../images/shopping-bag.png";
import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import { useStyles } from "../../styles/UpperNav";
import { SpanModal } from "../../../src/containers/commons/SpanModal";
import SignUpPage from "../../containers/customers/SignupPage";
import LoginPage from "../../containers/customers/LoginPage";

const UpperNav = ({ isLoggedIn, allDepartments, totalAmount }) => {
  const classes = useStyles();
  const { departments } = allDepartments;
  return (
    <Fragment>
      <nav className="navbar navbar-expand-md sticky-top navbar-light bg-light pt-0 pb-0 fir">
        <div className="navbar-brand">
          Hey!{" "}
          <a className="text-danger" href="#">
            <SpanModal
              spanText="Sign In"
              modalTitle="Login To Your ShopMate Account"
              modalWidth="450px"
              modalContent={<LoginPage />}
              className={classes.navItems}
            />
          </a>{" "}
          or{" "}
          <a href="#" className="text-danger">
            <SpanModal
              spanText="Register"
              modalTitle="Register For A ShopMate"
              modalWidth="450px"
              modalContent={<SignUpPage />}
              className={classes.navItems}
            />
          </a>
        </div>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample04"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample04">
          <ul className="navbar-nav m-auto nav-list">
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Daily Deals <span className="sr-only">(current)</span>
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Sell
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link" href="#">
                Help & Contact
              </a>
            </li>
          </ul>
          <form className="form-inline my-2 my-md-0">
            <NavLink to="/mycart/" className="my-bag">
              <img src={bag} alt="bag" /> &nbsp; Your Bag &nbsp;{" "}
              <span className="text-danger"> $ {totalAmount}</span>
            </NavLink>
          </form>
        </div>
      </nav>
      <nav className="navbar navbar-expand-md fixed-top navbar-dark bg-dark nav-margin">
        <a className="navbar-brand logo" href="#">
          <NavLink to="/" className={classes.shopmateLogo}>
            SHOPMATE
          </NavLink>
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExample05"
          aria-controls="navbarsExample05"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExample05">
          <ul className="navbar-nav m-auto">
            {departments &&
              departments.map(({ department_id, name }) => (
                <Fragment key={department_id}>
                  <NavLink
                    to={`/department/${department_id}`}
                    className={classes.navItemsMainBar}
                  >
                    <li className="nav-item active">
                      <a className="nav-link nav-link-item" href="#">
                        {name}
                        <span className="sr-only">(current)</span>
                      </a>
                    </li>
                  </NavLink>
                  &nbsp;&nbsp;&nbsp;&nbsp;
                </Fragment>
              ))}
          </ul>
          <form className="form-inline my-2 my-md-0 text-white">
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput
                }}
              />
            </div>
          </form>
        </div>
      </nav>
    </Fragment>
  );
};
export default UpperNav;
