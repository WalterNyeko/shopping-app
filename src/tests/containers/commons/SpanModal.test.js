import React from "react";
import { SpanModal } from "../../../containers/commons/SpanModal";
import { mount } from "enzyme";

describe("SpanModal", () => {
  const wrapper = mount(<SpanModal />);
  it("should render modal without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show modal when its span is clicked", () => {
    expect(wrapper.instance().state.visible).toEqual(false);
    const theSpan = wrapper.find("span").at(0);
    theSpan.simulate("click");
    expect(wrapper.instance().state.visible).toEqual(true);
  });
});
