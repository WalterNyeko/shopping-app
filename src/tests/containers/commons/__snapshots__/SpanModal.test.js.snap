// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanModal should render modal without crashing 1`] = `ReactWrapper {}`;
